src/ChatPage.cpp

```diff
@@ -167,6 +167,9 @@ ChatPage::ChatPage(QSharedPointer<UserSettings> userSettings, QObject *parent)
             view_manager_,
             &TimelineViewManager::initializeRoomlist);
     connect(this, &ChatPage::syncUI, this, [this](const mtx::responses::Sync &sync) {
+        // The timestamp of the first time we have sync the process:
+        // don't send desktop notifications for events who happened before
+        static QDateTime firstSyncTimestamp = QDateTime::currentDateTime();
```

Comment:
This wouldn't work across restarts, I think. I suggest you just set a bool flag based on if it is the first sync after startup or not. You can even just pass that in the signal.

Reply:
Actually, the bool solution was my first implementation, and I found out it doesn't work that well. It seems that sometimes, the messages are synced on the second sync, not on the first.

I'm not sure what you mean by "This wouldn't work across restarts". The idea is simple: there shouldn't be desktop notifications for events that happened when Nheko wasn't running. So if you close Nheko and open it later, there won't be desktop notifications for the events that happened while it was closed. It is intended, and as far as I know most messengers work that way.

Reply:
I used the wrong words. If you log out and back in, this probably wouldn't do the right thing.

Reply:
In my opinion, the "right thing" is to send desktop notifications when Nheko is running and the user is connected, and to not send them when it is not. It seems that in the current state of Nheko, the desktop notifications aren't sent for events that happened while you were disconnected, that is, when you log in, you don't get desktop notifications for unread messages. This is a correct behavior in my opinion, and I extend this behavior for events happening while Nheko isn't running.

Reply:
I still think using a static timestamp is the wrong solution. For example you will get a similar notification spam after you suspended your laptop for a long time. Could we maybe instead store the timestamp of the last successful sync and if the difference is larger than a few hours, you calculate a summary notification? If the timestamp defaults to unix epoch, it would always skip the sync after startup as well. Alternatively we could switch to summary notifications if there would be more than a certain number of notifications (like 5 or so). Then we would need no timestamp at all and it would get rid of the notification spam in general as well, since that is the primary issue you were trying to address, right?

Reply:
Unfortunately, the last successful sync timestamp solution wouldn't work for the same reason as the bool solution can't: sometimes, the messages are synced on the second sync, not on the first, and in this case the notification swarm isn't prevented.

The notification threshold solution is a good idea, I'll try it out. I'm not fond of arbitrary numbers, however, maybe we should add a setting? For now I'll stick to 5, I guess.

Reply:
I usually prefer hardcoding it first, then only adding an option if we can't find a good value for everyone. Every option is another option to test :)